astropy/cosmology/core.py

```diff
@@ -297,12 +297,8 @@ def __init__(self, H0, Om0, Ode0, Tcmb0=0, Neff=3.04,
         if self._nneutrinos > 0 and self._Tcmb0.value > 0:
             self._neff_per_nu = self._Neff / self._nneutrinos
 
-            # We can't use the u.Quantity constructor as we do above
-            # because it doesn't understand equivalencies
-            if not isinstance(m_nu, u.Quantity):
-                raise ValueError("m_nu must be a Quantity")
-
-            m_nu = m_nu.to(u.eV, equivalencies=u.mass_energy())
+            with u.set_enabled_equivalencies(u.mass_energy()):
+                m_nu = u.Quantity(m_nu, u.eV)
```

Comment:
Hm, this would allow numbers through - is that intended? That seems to be an API change. Maybe we want this instead
```suggestion
                m_nu = u.Quantity(m_nu).to(u.eV)
```
?

Reply:
While I would personally prefer to limit this to quantity-like and not allow numbers, this actually brings ``m_nu`` more in line with the rest of the cosmology package, where ``H0`` and ``Tcmb0`` both allow non-unit-ful inputs.